drivers/serial/uart_ns16550_port_x.h

```diff
@@ -11,11 +11,11 @@
 static void irq_config_func_@NUM@(struct device *port);
 #endif
 
-static const struct uart_ns16550_device_config uart_ns16550_dev_cfg_@NUM@ = {
-	.sys_clk_freq = DT_UART_NS16550_PORT_@NUM@_CLK_FREQ,
+static const struct uart_ns16550_config uart_ns16550_dev_cfg_@NUM@ = {
```

Comment:
why was this and struct uart_ns16550_dev_data_t renamed?

Reply:
This and uart_ns16550_dev_data_t were renamed to be consistent with an example of an already implemented one such as esp32 uart driver (https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/serial/uart_esp32.c)

Reply:
Thanks for pointing it out, I can revert my change and push back up here to remove any confusion. It builds without issue (just tried it), but still gets stuck booting in qemu.

Reply:
for renaming/whitespace changes these are fine just be sure to split them out into a separate patch in the series, it makes the PR easier to review